game/missiongenerator/convoygenerator.py

```diff
@@ -38,11 +39,28 @@ def generate_convoy(self, convoy: Convoy) -> VehicleGroup:
             convoy.units,
             convoy.player_owned,
         )
-        group.add_waypoint(
-            convoy.route_end,
-            speed=kph(40).kph,
-            move_formation=PointAction.OnRoad,
-        )
+
+        if self.game.settings.perf_moving_convoys:
+            group.add_waypoint(
+                convoy.route_end,
+                speed=kph(40).kph,
+                move_formation=PointAction.OnRoad,
+            )
+        else:
+            # perf_moving_convoys is disabled, so add the convoy between the route start and route end.
+            # This option aims to remove long routes for ground vehicles between control points,
+            # since the CPU load for long routes on DCS is pretty heavy.
+            frontline = FrontLine(convoy.origin, convoy.destination)
+
+            # Select the first route segment from the origin towards the destination
+            # so the convoy spawns at the origin CP. This allows the convoy to be
+            # targeted by BAI flights and starts it within the protection umbrella of the CP.
+            group.add_waypoint(
+                frontline.segments[0].point_b,
```

Comment:
There's no guarantee that this is any shorter than `convoy.route_end`, is there?

Also, refactor. You've duplicated everything but this argument.

```python
if self.game.settings.perf_moving_convoys:
    end_point = convoy.route_end
else:
    end_point = frontline.segments[0].point_b
```

Reply:
Done. No guarantees of course, but I haven't seen a campaign where the supply routes would only contain one waypoint.